libs/langchain/langchain/retrievers/google_cloud_enterprise_search.py

```diff
@@ -151,6 +166,28 @@ def _convert_search_response(
 
         return documents
 
+    def _convert_structured_search_response(
+        self, results: Sequence[SearchResult]
+    ) -> List[Document]:
+        """Converts a sequence of search results to a list of LangChain documents."""
+        from google.protobuf.json_format import MessageToDict
+
+        documents: List[Document] = []
+
+        for result in results:
+            document_dict = MessageToDict(
+                result.document._pb, preserving_proto_field_name=True
+            )
+
+            documents.append(
+                Document(
+                    page_content=str(document_dict.get("struct_data", {})),
+                    metadata={"id": document_dict["id"], "name": document_dict["name"]},
+                )
+            )
+
+        return documents
+
```

Comment:
I still think this could be simplified/modularized further, looks like the cycling through documents/converting to Dict is repeated for both structured and unstructured data.

Reply:
Hey Holt, I tried to do the change and I reverted back as I am not convinced its worth it. Apart from the loop everything inside is treated differently. additionally for structured data each "result" in the loop corresponds to one document. In unstructured data every result can have a set of chunks and each chunk corresponds to a document. the ideology on how you process results is different and I do not see the point of forcing the split of the code for the loop. + readability. Please reconsider